aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,486 @@
+- Less markup in source code
+- Strong typing with the model property. If the name of the property changes you'll get an error similar to the following:
+
+.. code-block:: HTML
+
+     An error occurred during the compilation of a resource required to process
+     this request. Please review the following specific error details and modify
+     your source code appropriately.
+    
+     Type expected
+      'SimpleViewModel' does not contain a definition for 'Email' and no
+      extension method 'Email' accepting a first argument of type 'SimpleViewModel'
+      could be found (are you missing a using directive or an assembly reference?)
+
+The Validation Tag Helpers
+---------------------------
+
+There are two Validation Tag Helpers. The `Validation Message Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/ValidationMessageTagHelper/index.html>`__ (which displays a validation message for a single property on your model), and the `Validation Summary Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/ValidationSummaryTagHelper/index.html>`__ (which displays a summary of validation errors). The `Input Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/InputTagHelper/index.html>`__ adds HTML5 client side validation attributes to input elements based on data annotation attributes on your model classes. The Validation Tag Helper displays these error messages when a validation error occurs. 
+
+The Validation Message Tag Helper
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The `Validation Message Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/ValidationMessageTagHelper/index.html>`__  is used with the ``asp-validation-for`` attribute on a HTML `span <https://developer.mozilla.org/en-US/docs/Web/HTML/Element/span>`__ element.
+
+.. code-block:: HTML
+  
+  <span asp-validation-for="Email">
+  
+The Validation Message Tag Helper will generate the following HTML:
+
+.. code-block:: HTML
+
+    <span class="field-validation-valid" 
+      data-valmsg-for="Email" 
+      data-valmsg-replace="true">
+
+The `Validation Message Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/ValidationMessageTagHelper/index.html>`__  adds the `HTML5 <https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5>`__  ``data-valmsg-for="property"`` attribute to the ``<span>`` element, which attaches the validation error messages on the input field of the specified model property with ``<span>`` element. When a client side validation error occurs, `jQuery <https://jquery.com/>`__ displays the error message in the ``<span>`` element.
```

Comment:
Also mention that "When a server-side validation error occurs e.g. a custom validation or when client-side validation is disabled, MVC places that error message as the body of the `<span>` element." or some such.